### ISSUE TITLE
TypeError: Cannot read property 'filesCollection' of undefined

### ISSUE THREAD

**Issue by rich-m21** [VeliovGroup/meteor-autoform-file#43]
I'm getting an error when I try to implement Ostrio-files, s3 and autoform directly with a QuickForm
I've done it on previous versions without hassle but on these:
ostrio:autoform-files@2.1.1
ostrio:cookies@2.3.0
ostrio:files@1.10.1
METEOR@1.7.0.3
Using Mac
Also using dburles:mongo-collection-instances

Client logs
```
Exception from Tracker recompute function:
meteor.js?hash=c8108d734cc548d91539b054e14b613d0c4512cc:1010 TypeError: Cannot read property 'filesCollection' of undefined
    at Blaze.TemplateInstance.<anonymous> (fileUpload.js:26)
    at blaze.js?hash=a1ff2d6d5ecd59ee11e2ba260b8650a9d1140f59:3398
    at Function.Template._withTemplateInstanceFunc (blaze.js?hash=a1ff2d6d5ecd59ee11e2ba260b8650a9d1140f59:3744)
    at fireCallbacks (blaze.js?hash=a1ff2d6d5ecd59ee11e2ba260b8650a9d1140f59:3394)
    at Blaze.View.<anonymous> (blaze.js?hash=a1ff2d6d5ecd59ee11e2ba260b8650a9d1140f59:3474)
    at fireCallbacks (blaze.js?hash=a1ff2d6d5ecd59ee11e2ba260b8650a9d1140f59:2014)
    at Object.Tracker.nonreactive (tracker.js:603)
    at blaze.js?hash=a1ff2d6d5ecd59ee11e2ba260b8650a9d1140f59:2011
    at Object.Blaze._withCurrentView (blaze.js?hash=a1ff2d6d5ecd59ee11e2ba260b8650a9d1140f59:2271)
    at Object.Blaze._fireCallbacks (blaze.js?hash=a1ff2d6d5ecd59ee11e2ba260b8650a9d1140f59:2010)
```
Thanks in advance.

**Comment by dr-dimitru** [comment]
@rich-m21 thank you for reporting this.
I can suggest only to try to use globally defined collection. Also you should make sure name of collection is the same as its variable.

@jankapunkt could you take a look on this, please :)

**Comment by jankapunkt** [comment]
@rich-m21 can you please also add the version number of simpl-schema or add a link to a minimal Repo that reproduced the error. I got this error some weeks ago but not sure if it was trivial or related to Meteor 1.7 or so.

**Comment by rich-m21** [comment]
Sorry, I'm also using simpl-schema@1.5.3

**Comment by jankapunkt** [comment]
There is a known issue with simpl-schema > 1.5 and quickform.

Does your issue resolve using simpl-schema 1.4.3?

```
meteor npm uninstall --save simpl-schema
meteor npm prune
meteor npm install --save simpl-schema@1.4.3
```

If this does not solve your issue I will try on some reproduction.

**Comment by dr-dimitru** [comment]
@jankapunkt thank you for quick response.
@rich-m21 please keep this thread updated.

**Comment by jankapunkt** [comment]
@rich-m21 did this solve your issue?

**Comment by rich-m21** [comment]
Hello guys,
Sadly no,
I ended up rolling back to, given to time constraints.
METEOR@1.6.0.1
===atmosphere===
aldeed:simple-schema@1.5.3
aldeed:collection2@2.10.0
aldeed:autoform@5.8.1
ostrio:autoform-files@1.0.12
ostrio:cookies@2.2.5
ostrio:files@1.9.7

Funny thing I asked a colleague of mine's and he managed to integrate it ok (just the individual upload file, not the auto form integration.I had no luck in neither.
Thanks for the quick response guys.

**Comment by rich-m21** [comment]
Yes, I'm aware of that.
What I meant to say is that he managed to integrate it not using autoform-files just the SEPARATE upload element with the custom upload helper. Not the autoform implementation.
Cheers

**Comment by jankapunkt** [comment]
ok I will create a reproduction using the given packages and versions from your initial comment

**Comment by jankapunkt** [comment]
Finally found something. @rich-m21 are you sure, that the name of the `collection` in the schema matches (case-sensitive) the name of the `collectionName` property of the FilesCollection?

When you define the FilesCollection with `collectionName : 'images'` and then refer via schema to `collection: 'Images'` it will throw the above mentioned error!

@dr-dimitru maybe I update the documentation in accordance to the names used in the `ostrio:files` wiki so that such confuison may not appear?

I could also catch the error and show a list of available collections using `Mongo.Collection.getAll()` before rethrow so in the error output you could easily see the collection you entered is in the list of all registered collections.

**Comment by dr-dimitru** [comment]
> @dr-dimitru maybe I update the documentation in accordance to the names used in the ostrio:files wiki so that such confuison may not appear?

Go ahead, PRs are always very welcome.

**Comment by dr-dimitru** [comment]
> When you define the FilesCollection with collectionName : 'images' and then refer via schema to collection: 'Images' it will throw the above mentioned error!

This was in my [first message](https://github.com/VeliovGroup/meteor-autoform-file/issues/43#issuecomment-410447192) isn't it? It may sound not that clear, I agree. Let's update the docs

**Comment by jankapunkt** [comment]
> This was in my first message isn't it? It may sound not that clear, I agree. Let's update the docs
Correct you already mentioned that. What I wanted to say is, that this is the only cause I found for this error at this line. Will clear that in the docs

**Comment by jankapunkt** [comment]
Please see #44 

Regarding documentation I just found one different use of `"Images"` as collection name in 

https://github.com/VeliovGroup/Meteor-Files/blob/5276d1ee352c1f64ee93968ad03b2dcad7453907/docs/gridfs-integration.md

where it is declared as `"images"` so I will create a PR there to fix this

**Comment by dr-dimitru** [comment]
Solved and published as [`v2.1.2`](https://github.com/VeliovGroup/meteor-autoform-file/releases/tag/2.1.2). Thank you both a lot.

Feel free to reopen it in case if the issue is still persists on your end.
